scripts/fgi.sh

```diff
@@ -0,0 +1,269 @@
+    sleep 10
+  done
+  echo "Failed to schedule job on a cluster as all are busy"
+  exit 1
+}
+
+# wait for the given pod to be scheduled
+kube_wait_pod () {
+  local pod_name context phase
+  pod_name="${1}"
+  context="${2}"
+  for i in {1..360} ; do
+    phase=$(kubectl --context="${context}" get pod "${pod_name}" -o jsonpath="{.status.phase}" || echo -n "kubectlfailed")
+    if [[ "${phase}" == "kubectlfailed" ]]; then
+        echo "kubectl get pod ${pod_name} failed. Retrying."
+        sleep 10
+        continue
+    fi
+
+    # pod is either Running, Succeeded, or Failed
+    # https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase
+    if [[ $phase != "Pending" &&  $phase != "Unknown" ]]; then
+      echo "${pod_name} reached phase: ${phase}"
+      return
+    fi
+
+    # error pulling the image
+    if kubectl --context="${context}" get pod "${pod_name}" | grep -i -e ImagePullBackOff -e InvalidImageName -e ErrImagePull &>/dev/null; then
+      image_name=$(kubectl --context="${context}" get pod "${pod_name}" -o jsonpath="{.spec.containers[0].image}")
+      echo "${pod_name} name failed to be scheduled because there was an error pulling the image: ${image_name}"
+      # Delete the pod so that it doesn't block other pods from being scheduled on this
+      kubectl --context="${context}" delete pod "${pod_name}"
+      exit 1
+    fi
+
+    echo "Waiting for ${pod_name} to be scheduled. Current phase: ${phase}"
+    sleep 10
+  done
+  echo "Pod ${pod_name} failed to be scheduled"
+  exit 1
+}
+
+while (( "$#" )); do
+  case "$1" in
+    -p|--pr)
+      PR=$2
+      shift 2
+      ;;
+    -L|--local-build)
+      LOCAL_BUILD="yes"
+      shift 1
+      ;;
+    --timeout-secs)
+      TIMEOUT_SECS=$2
+      shift 2
+      ;;
+    --local-swarm)
+      LOCAL_SWARM="yes"
+      shift 1
+      ;;
+    -T|--tag)
+      TAG=$2
+      shift 2
+      ;;
+    -W|--workspace)
+      WORKSPACE=$2
+      shift 2
+      ;;
+    -E|--env)
+      ENV="$ENV $2"
+      shift 2
+      ;;
+    *) # end argument parsing
+      break
+      ;;
+  esac
+done
+
+if [ -n "$LOCAL_SWARM" ]; then
+  cargo run -p forge -- --local-swarm
+  exit $EXIT_CODE
+fi
+
+if ! which kubectl &>/dev/null; then
+  echo "kubectl is not installed. Please install kubectl. On mac, you can use: brew install kubectl"
+  echo "scripts/dev_setup.sh -i kubectl"
+  exit 1
+fi
+
+# build and push the images to be used
+if [ -z "$TAG" ]; then
+  if [ -z "$LOCAL_BUILD" ]; then
+    aws codebuild list-projects >/dev/null || (echo "Failed to access codebuild, try awsmfa?"; exit 1)
+    ./docker/build-aws.sh --build-forge --version "pull/${PR}"
+    TAG="dev_${USER}_pull_${PR}"
+    echo "**TIP Use fgi -T $TAG <...> to restart this run with same tag without rebuilding it"
+  else
+    TAG="dev_$(whoami)_$(git rev-parse --short HEAD)"
+    TAG=$TAG ./docker/build-push-local.sh
+  fi
+fi
```

Comment:
This is kinda what I mean about public.  Should this be somewhere else if it's referencing `awsmfa`?